docs/config-file.md

```diff
@@ -119,21 +119,50 @@ These configurations apply to Glee itself, rather than any specific protocol.
 
 |Field|Description|
 |---|---|
-|mqtt.authentication| MQTT authentication configuration|
-|mqtt.authentication.cert| Client certificate
-|mqtt.authentication.clientId| MQTT client Id for authentication
-|mqtt.authentication.userPassword| username and password parameters for authentication|
-|mqtt.authentication.userPassword.username| username parameter
-|mqtt.authentication.userPassword.password| password parameter
+|mqtt.auth| MQTT authentication configuration|
+|mqtt.auth.cert| Client certificate
+|mqtt.auth.clientId| MQTT client Id for authentication
+|mqtt.auth.username| username parameter
+|mqtt.auth.password| password parameter
 
 ### Kafka
 
 |Field|Description|
 |---|---|
-|kafka.authentication| Kafka authentication configuration|
-|kafka.authentication.key | Kafka Broker Key
-|kafka.authentication.cert| Client certificate
-|kafka.authentication.clientId| Kafka client Id for authentication
-|kafka.authentication.rejectUnauthorized | Boolean flag for accepting the valid SSL certificates
-|kafka.authentication.username| The username to use during authentication.
-|kafka.authentication.password| The password to use during authentication.
\ No newline at end of file
+|kafka.auth| Kafka authentication configuration|
+|kafka.auth.key | Kafka Broker Key
+|kafka.auth.cert| Client certificate
+|kafka.auth.clientId| Kafka client Id for authentication
+|kafka.auth.rejectUnauthorized | Boolean flag for accepting the valid SSL certificates
+|kafka.auth.username| The username to use during authentication.
+|kafka.auth.password| The password to use during authentication.
+
+
+### Auth Config
+
+Most client like `ws`,`kafka` and `mqtt` has auth config which is used for passing auth parameters, all these config could be an object or an function that returns the specific object defined by each protocol.
```

Comment:
```suggestion
Most clients like `ws`,`kafka`, and `mqtt` have `auth` fields that are used for passing auth parameters. All these configurations can be an object or a function that returns the specific object defined by each protocol.
```

Reply:
Suggestion: go to https://grammarly.com and create a free account. Then install the browser extension and the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=znck.grammarly). It will help you write better :)